crypto/ml_dsa/ml_dsa_sign.c

```diff
@@ -310,53 +314,33 @@ static int ml_dsa_verify_internal(const ML_DSA_KEY *pub, int msg_is_mu,
  *
  * ML_DSA pure signatures are encoded as M' = 00 || ctx_len || ctx || msg
  * Where ctx is the empty string by default and ctx_len <= 255.
+ * The message is appended to the encoded context later so it is not needed
+ * in this function.
  *
  * Note this code could be shared with SLH_DSA
  *
- * @param msg A message to encode
- * @param msg_len The size of |msg|
  * @param ctx An optional context to add to the message encoding.
  * @param ctx_len The size of |ctx|. It must be in the range 0..255
- * @param encode Use the Pure signature encoding if this is 1, and dont encode
- *               if this value is 0.
- * @param tmp A small buffer that may be used if the message is small.
- * @param tmp_len The size of |tmp|
- * @param out_len The size of the returned encoded buffer.
- * @returns A buffer containing the encoded message. If the passed in
- * |tmp| buffer is big enough to hold the encoded message then it returns |tmp|
- * otherwise it allocates memory which must be freed by the caller. If |encode|
- * is 0 then it returns |msg|. NULL is returned if there is a failure.
+ * @param encoded A small buffer to store the encoded context.
+ * @param encoded_size The size of |encoded|
+ * @param out_len The size of the resulting encoded buffer.
```

Comment:
Returns the size of

Reply:
This line is unchanged from the original, it is just the diff that makes it look not so.